bravado/mapping/request.py

```diff
@@ -0,0 +1,60 @@
+from bravado.mapping.operation import log
+from bravado.mapping.param import unmarshal_param
+
+
+class RequestLike(object):
+    """
+    Define a common interface for bravado to interface with server side
+    request objects.
+
+    Subclasses are responsible for providing attrs for __required_attrs__.
+    """
+    __required_attrs__ = [
+        'path',     # dict of URL path parameters
+        'params',   # dict of parameters from the query string and request body.
```

Comment:
This shouldn't reflect pyramid in any way. This interface is a bravado-core interface. We can use an adapter to make the pyramid request fit what bravado needs, like https://github.com/striglia/pyramid_swagger/blob/fe3b87706673c58929e2a2b0a9d3236f4e4517ca/pyramid_swagger/tween.py#L132

I actually think we can use that exact adapter implementation.

The adapter would use `request.GET`. You don't need to know the request method, because post always uses body which will be swagger param type of `body` or `form`.